feature(Set Number of months for CMM)

lomamech opened the pull request:
- Add in Enterprise module the possibility for the user to set number of
  months used to the calcul of CMM

- Update stock core and using the number from enterprise setting
- (Test) add property month_average_consumption in Integration test

closes #4151

lomamech committed:
feature(Set Number of months for CMM)
- Add in Enterprise module the possibility for the user to set number of
  months used to the calcul of CMM

- Update stock core and using the number from enterprise setting
- (Test) add property month_average_consumption in Integration test

jniles commented on client/src/modules/enterprises/enterprises.html:
```diff
@@ -217,6 +217,18 @@
+                  FORM.LABELS.DEFINE_MONTH_AVERAGE_CONSUMPTION
+                </label>
+                <input type="number" class="form-control" autocomplete="off"
+                  name="month_average_consumption" required ng-model="EnterpriseCtrl.enterprise.settings.month_average_consumption"
```
Does it need to be required?  Or do you set it to a default value in the enterprise controller?  There are some institutions that won't use stock management, so we shouldn't force them to set this variable unnecessarily.

jniles commented on server/models/migrations/next/migrate.sql:
```diff
@@ -1,3 +1,5 @@
 Migrate to next
  */
+-- @lomamech 2020-03-24
+ALTER TABLE `enterprise_setting` ADD COLUMN `month_average_consumption` SMALLINT(5) NOT NULL DEFAULT 6;
```
👍

jniles commented on test/integration/enterprises.js:
```diff
@@ -28,7 +28,7 @@ describe('(/enterprises) Enterprises API', () => {
     location_id : '1f162a10-9f67-4788-9eff-c1fea42fcc9b',
     currency_id : 1,
-    settings : { enable_price_lock : 0 },
+    settings : { enable_price_lock : 0, month_average_consumption : 10 },
```
Can you check that the GET request returns this value?

Can you also write a test that shows it properly sets the default value?

jniles reviewed (commented):
This looks good to me.  The only change I would like to see are the two integration tests:
  1. Check to see that month_average_consumption is being read from the database on GET.
  2. Make sure a default value is set when you create an enterprise.

Otherwise, looks excellent!  This will be a huge help.

lomamech committed:
test(Enterprise: Month Average Consuption)
- Adding test for check default value and Updating for
  month_average_consuption`

closes #4151
